tests/spec/core/markdown-spec.js

```diff
@@ -22,7 +22,7 @@ describe("Core - Markdown", () => {
     });
     const foo = doc.getElementById("foo");
     expect(foo).toBeTruthy();
-    expect(foo.textContent).toBe("1. Foo");
+    expect(foo.textContent.trim()).toBe("1. Foo");
```

Comment:
Now that we don't throw out text nodes, I think the markdown parser generates some extraneous whitespace after the headings. No big deal though.

Reply:
Should investigate I think.

Reply:
Yeah, we have some inconsistency on how the ids are being assigned to sections and headers somewhere... I'll investigate further.

That also explains `.trim()` in a few places above... the `<section>`s are now getting the ids that were on the heading elements.